ts/src/base/Exchange.ts

```diff
@@ -1362,6 +1390,50 @@ export default class Exchange {
     // ------------------------------------------------------------------------
     // METHODS BELOW THIS LINE ARE TRANSPILED FROM JAVASCRIPT TO PYTHON AND PHP
 
+    checkProxySettings (url, method, headers, body) {
+        let proxyUrl = (this.proxyUrl !== undefined) ? this.proxyUrl : this.proxy_url;
+        if (typeof proxyUrl === 'function') {
+            proxyUrl = proxyUrl (url, method, headers, body);
+        }
+        let httpProxy = (this.httpProxy !== undefined) ? this.httpProxy : this.http_proxy;
+        if (typeof httpProxy === 'function') {
+            httpProxy = httpProxy (url, method, headers, body);
```

Comment:
@ttodua question, do we need to support this function format?

Reply:
in previous version we supported callbacks for `.proxy` property. so, i've continued support for it, not to make a regress.

so, it's sometimes kinda useful for users, to set rotating proxies or etc, i.e.
```
this.httpProxy = function ( ...) {
  i ++;
  return myProxies[i];
};
```
or whatever their criterias.

Reply:
@ttodua I understand but personally I don't love this approach because it's hard to type a prop that can be either a string or a function that we don't know exactly at compile time which argument it takes and what it returns. Second, assuming that we keep it like that it seems we're not keeping the backwards-compability for the `this.proxy` prop, we only check it in the line 1426 and we don't check its type

Reply:
to start from 2nd  thing : `this.proxy` is backward supported now, as it is just assigned to `this.proxyUrl` which is later checked for type.
about first thing - ok, if you think we shouldn't do this way, i'll change and re-push

Reply:
@ttodua if you don't mind and agree with this, I would probably type them as strings and remove the callback functionality

Reply:
sure, I've been working on that and finished. pushed the changes, review now.